src/contracts/GPv2Settlement.sol

```diff
@@ -126,10 +126,32 @@ contract GPv2Settlement {
     }
 
     /// @dev Invalidate onchain an order that has been signed offline.
-    /// @param orderDigest The unique digest associated to the parameters of an
-    /// order. See [`orderUid`] for details.
-    function invalidateOrder(bytes32 orderDigest) public {
-        filledAmount[orderUid(orderDigest, msg.sender)] = uint256(-1);
+    /// @param orderUid The unique identifier of the order that is to be made
+    /// invalid after calling this function. The user that created the order
+    /// must be the the sender of this message. See [`extractOrderUidParams`]
+    /// for details on orderUid.
+    function invalidateOrder(bytes calldata orderUid) external {
```

Comment:
Not sure if this is a good idea, but given that we later `require(orderUid.length == 56);` could this also become `byte[56] calldata orderUid`?

Reply:
In general `byte[]` is to be avoided because each byte is padded to a full byte32 ([source](https://docs.soliditylang.org/en/v0.7.0/types.html#bytes-and-strings-as-arrays)).
I tested just to know if this was the case for a byte array of bounded size (`function test(bytes[2] calldata in) external`) and the data field of a transaction calling this (`contract.test(['0x11', '0x22'])`) is very much inefficient:
```
0x898c23d0
0000000000000000000000000000000000000000000000000000000000000020
0000000000000000000000000000000000000000000000000000000000000040
0000000000000000000000000000000000000000000000000000000000000080
0000000000000000000000000000000000000000000000000000000000000001
1100000000000000000000000000000000000000000000000000000000000000
0000000000000000000000000000000000000000000000000000000000000001
2200000000000000000000000000000000000000000000000000000000000000
```

Reply:
Does `bytes56` exist?